internal/cache/file/cache_handle.go

```diff
@@ -148,28 +187,27 @@ func (fch *CacheHandle) Read(ctx context.Context, object *gcs.MinObject, offset
 	}
 
 	// We are here means, we have the data downloaded which kernel has asked for.
-	_, err = fch.fileHandle.Seek(offset, 0)
-	if err != nil {
-		errMsg := fmt.Sprintf("%s: while seeking for %d offset in local file: %v", util.ErrInSeekingFileHandleMsg, offset, err)
-		return 0, errors.New(errMsg)
-	}
-
-	n, err = io.ReadFull(fch.fileHandle, dst)
-	// Returns io.ErrUnexpectedEOF when buffer size is more than content to be read.
+	n, err = fch.fileHandle.ReadAt(dst, offset)
+	requestedNumBytes := int(requiredOffset - offset)
+	// dst buffer has fixed size of 1 MiB even when the offset is such that
+	// offset + 1 MiB > object size. In that case, io.ErrUnexpectedEOF is thrown
+	// which should be ignored.
 	if err == io.EOF || err == io.ErrUnexpectedEOF {
 		err = nil
 	}
 	if err != nil {
 		errMsg := fmt.Sprintf("%s: while reading from %d offset of the local file: %v", util.ErrInReadingFileHandleMsg, offset, err)
 		return 0, errors.New(errMsg)
+	} else if n != requestedNumBytes {
```

Comment:
Sorry for the to and fro, I think, this check should be inside `line:195 if block`?